python/cudf/cudf/core/indexing.py

```diff
@@ -168,13 +168,18 @@ def _loc_to_iloc(self, arg):
         from cudf.core.series import Series
 
         if is_scalar(arg):
-            try:
-                found_index = self._sr.index._values.find_first_value(
-                    arg, closest=False
-                )
+            if isinstance(arg, int):
```

Comment:
if arg is a `cudf.Scalar` this won't work as expected

Reply:
Q:What is  the expected behavior when indexing with a` cudf.scalar ` value?

Reply:
Not sure what is the expected behavior when indexer is a cudf Scalar? I don't think it's even possible to build a
Series with cudf Scalar index? so how can the indexing value possibly be a cudf Scalar?

In [18]: index
Out[18]: [Scalar(2, dtype=int32), Scalar(4, dtype=int32)]

In [19]: ps = pd.Series([1, 2], index=pd.Index(index))

In [20]: ps
Out[20]: ---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.PyObjectHashTable.map_locations()

TypeError: unhashable type: 'Scalar'
Exception ignored in: 'pandas._libs.index.IndexEngine._call_map_locations'
Traceback (most recent call last):
  File "pandas/_libs/hashtable_class_helper.pxi", line 4588, in pandas._libs.hashtable.PyObjectHashTable.map_locations
TypeError: unhashable type: 'Scalar'

Scalar(2, dtype=int32)    1
Scalar(4, dtype=int32)    2
dtype: int64